pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/gates/tests/Test_TNCuda_MPO.cpp

```diff
@@ -0,0 +1,226 @@
+// Copyright 2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the License);
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an AS IS BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <complex>
+#include <vector>
+
+#include <catch2/catch.hpp>
+
+#include "DevTag.hpp"
+#include "MPSTNCuda.hpp"
+#include "TNCudaGateCache.hpp"
+
+#include "TestHelpers.hpp"
+#include "TestHelpersTNCuda.hpp"
+
+/// @cond DEV
+namespace {
+using namespace Pennylane::LightningGPU;
+using namespace Pennylane::LightningTensor;
+using namespace Pennylane::LightningTensor::TNCuda::Gates;
+using namespace Pennylane::Util;
+namespace cuUtil = Pennylane::LightningGPU::Util;
+using namespace Pennylane::LightningTensor::TNCuda::Util;
+} // namespace
+/// @endcond
+
+TEMPLATE_LIST_TEST_CASE("TNCuda::applyMPO::2+_wires", "[TNCuda_Nonparam]",
+                        TestMPSBackends) {
+    const bool inverse = GENERATE(false, true);
+    using TNDevice_T = TestType;
+    using cp_t = typename TNDevice_T::ComplexT;
+    using PrecisionT = typename TNDevice_T::PrecisionT;
+
+    std::size_t num_qubits = 3;
+    std::size_t maxExtent = 2;
+    std::size_t max_mpo_bond = 16;
+    DevTag<int> dev_tag{0, 0};
+
+    std::unique_ptr<TNDevice_T> tn_state =
+        createTNState<TNDevice_T>(num_qubits, maxExtent);
+
+    std::vector<std::vector<cp_t>> mpo_cnot(2,
+                                            std::vector<cp_t>(16, {0.0, 0.0}));
+
+    // in-order decomposition of the cnot operator
+    // data from scipy decompose in the lightning.tensor python layer
```

Comment:
can you include a little snippet or functions used to obtain these?

Reply:
It's a good idea, but I would like to put this suggestion on hold since we're going to get all `svd` decomposition operations done with C++ next Q. We will remove these tests later. `#TODO` mark added.